OpenRA.Mods.Common/Traits/SupportPowers/SpawnActorPower.cs

```diff
@@ -50,30 +62,113 @@ public override void Activate(Actor self, Order order, SupportPowerManager manag
 			base.Activate(self, order, manager);
 
 			var info = Info as SpawnActorPowerInfo;
+			var position = order.Target.CenterPosition;
+			var cell = self.World.Map.CellContaining(position);
+
+			if (!self.World.Map.Contains(cell))
+				return;
+
+			if (!info.AllowUnderShroud && self.World.ShroudObscures(cell))
+				return;
 
-			if (info.Actor != null)
+			if (info.Terrain != null && !info.Terrain.Contains(self.World.Map.GetTerrainInfo(cell).Type))
+				return;
+
+			if (info.Actor == null)
+				return;
```

Comment:
That field has `FieldLoader.Require` (and an `ActorReference`). Can this even be null?

Reply:
No. You are right. I removed it.